### PR TITLE
Added 'view sheds'

### PR DESCRIPTION
**Pull request by peterAdams1014**
Reworked some UD rendering functionality inside vcRender to accommodate requirements.

### CODE REVIEW

File: src/gl/metal/vcRenderShaders.cpp [hunk 1]
```diff
@@ -3,6 +3,8 @@
 const char *const g_VisualizationFragmentShader = "visualizationFragmentShader";
 const char *const g_VisualizationVertexShader = "visualizationVertexShader";
 
+const char *const g_ViewShedFragmentShader = "flatColorFragmentShader"; // not implemented
+const char *const g_ViewShedVertexShader = "udVertexShader"; // not implemented
```
Comment (OsiJr) [inline src/gl/metal/vcRenderShaders.cpp]: I think you can just say `nullptr` for these?

File: src/gl/opengl/vcRenderShaders.cpp [hunk 2]
```diff
@@ -126,10 +126,10 @@ void main()
 
   vec4 fragWorldPosition = u_inverseViewProjection * vec4(flippedUV * vec2(2.0) - vec2(1.0), depth * 2.0 - 1.0, 1.0);
   fragWorldPosition /= fragWorldPosition.w;
-
+  
   col.xyz = colourizeByHeight(col.xyz, fragWorldPosition.xyz);
   col.xyz = colourizeByDepth(col.xyz, depth);
-
+
```
Comment (OsiJr) [inline src/gl/opengl/vcRenderShaders.cpp]: EditorConfig didn't correct this?

File: src/vcRender.cpp [hunk 3]
```diff
@@ -665,6 +711,69 @@ void vcRender_VisualizationPass(vcState *pProgramState, vcRenderContext *pRender
...
+
+  vcGLState_SetDepthStencilMode(vcGLSDM_Always, false);
+  vcGLState_SetBlendMode(vcGLSBM_Additive);
+
+  pRenderContext->shadowShader.params.inverseProjection = udFloat4x4::create(udInverse(pProgramState->pCamera->matrices.projection));
+
+  if (pRenderContext->viewShedRenderingContext.pRenderView == nullptr)
+    vdkRenderView_Create(pProgramState->pVDKContext, &pRenderContext->viewShedRenderingContext.pRenderView, pRenderContext->udRenderContext.pRenderer, ViewShedMapRes.x / ViewShedMapCount, ViewShedMapRes.y);
+
+  for (size_t v = 0; v < renderData.viewSheds.length; ++v)
```
Comment (OsiJr) [inline src/vcRender.cpp]: Any plans to cache these to avoid the UD rerender/uploads if not dirty?

File: src/gl/metal/vcRenderShaders.cpp (hunk 1 above)
Comment (peterAdams1014) [reply src/gl/metal/vcRenderShaders.cpp]: I was just trying to get these to compile for Metal. For this PR it's not an issue if this breaks Metal as we're having other issues there. @DamianMadden thoughts on what to do here?

File: src/vcRender.cpp (hunk 3 above)
Comment (peterAdams1014) [reply src/vcRender.cpp]: Task added.
AB#437

File: src/gl/metal/vcRenderShaders.cpp (hunk 1 above)
Comment (DamianMadden) [reply src/gl/metal/vcRenderShaders.cpp]: Well this shouldn't cause a problem unless the functions are used

**Pull request merged**